### PR TITLE
Check access times in RankingTest.page_movement test

### PR DESCRIPTION
**Pull request by bratpiorka**
### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [ ] Other

### Checklist

- [ ] Code compiles without errors
- [ ] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [ ] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [ ] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/785)

### CODE REVIEW

File: test/memkind_memtier_data_movement_test.cpp [hunk 1]
```diff
@@ -24,6 +24,26 @@
...
 #include <test/proc_stat.h>
 
+std::chrono::duration<double> bench(size_t bench_num, size_t alloc_size,
+                                    void *start_addr)
+{
+    auto start = std::chrono::steady_clock::now();
+
+    long int temp = 0;
+    for (size_t i = 0; i < bench_num; i++) {
+        for (size_t j = 0; j < (alloc_size - 1) / sizeof(int); j++) {
```
Comment (maciekpac) [inline test/memkind_memtier_data_movement_test.cpp:34]: Maybe some bigger stride would be better ? E.g. :
```C
for (size_t j = 0; j < (alloc_size - 1) / sizeof(int); j+=32) {
```

For cache line size: 128 B, sizeof(int) == 4 ; 128/4 = 32

The best scenario would be the one that maximizes PMEM_ACCESS_TIME/DRAM_ACCESS_TIME ratio.

**Review by PatKamin — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/785)* status: all files reviewed, 6 unresolved discussions (waiting on @bratpiorka and @maciekpac)

___
*[test/memkind_memtier_data_movement_test.cpp, line 27 at r1](https://reviewable.io/reviews/memkind/memkind/785#-MzOVYsY0X2_I-zbTvwY:-MzOVYsY0X2_I-zbTvwZ:b6dmwmd) ([raw file](https://github.com/memkind/memkind/blob/3965cce5c8a5265a4f713a93a7acb2d92b0b72c7/test/memkind_memtier_data_movement_test.cpp#L27)):*
> ```cpp
> #include <test/proc_stat.h>
> 
> std::chrono::duration<double> bench(size_t bench_num, size_t alloc_size,
> ```

As this function is used only in one test (which is implemented 1500 lines from here), please create a class for the test case and move the bench() function there.
___
*[test/memkind_memtier_data_movement_test.cpp, line 1548 at r1](https://reviewable.io/reviews/memkind/memkind/785#-MzKzIrq6l8WkFf4JxEF-r1-1548:-MzOWAtkEhclWTygoOHr:b32v57) ([raw file](https://github.com/memkind/memkind/blob/3965cce5c8a5265a4f713a93a7acb2d92b0b72c7/test/memkind_memtier_data_movement_test.cpp#L1548)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

Maybe we could add a fixture that would map & memset in SetUp, unmap in TearDown ? The class could also keep the ```bench``` function as a class member. 

This was the preferred way of doing things, as we would like to keep this 1.6k long file as easy to maintain as possible and, in the future, make it easy to split the file if needed.
</blockquote></details>

+1

**Review by bratpiorka — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/785)* status: all files reviewed, 6 unresolved discussions (waiting on @maciekpac and @PatKamin)

___
*[test/memkind_memtier_data_movement_test.cpp, line 27 at r1](https://reviewable.io/reviews/memkind/memkind/785#-MzOVYsY0X2_I-zbTvwY:-MzP7GWz46QWNT5Nu1AR:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/3965cce5c8a5265a4f713a93a7acb2d92b0b72c7/test/memkind_memtier_data_movement_test.cpp#L27)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

As this function is used only in one test (which is implemented 1500 lines from here), please create a class for the test case and move the bench() function there.
</blockquote></details>

Done.
___
*[test/memkind_memtier_data_movement_test.cpp, line 34 at r1](https://reviewable.io/reviews/memkind/memkind/785#-MzKzIrq6l8WkFf4JxEF-r1-34:-MzP7HU8D2AsqYZR11Jd:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/3965cce5c8a5265a4f713a93a7acb2d92b0b72c7/test/memkind_memtier_data_movement_test.cpp#L34)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

Maybe some bigger stride would be better ? E.g. :
```C
for (size_t j = 0; j < (alloc_size - 1) / sizeof(int); j+=32) {
```

For cache line size: 128 B, sizeof(int) == 4 ; 128/4 = 32

The best scenario would be the one that maximizes PMEM_ACCESS_TIME/DRAM_ACCESS_TIME ratio.


</blockquote></details>

Done.
___
*[test/memkind_memtier_data_movement_test.cpp, line 36 at r1](https://reviewable.io/reviews/memkind/memkind/785#-MzKzIrq6l8WkFf4JxEF-r1-36:-MzP7IA2CFaAd7dus_OT:b-7m4ph3) ([raw file](https://github.com/memkind/memkind/blob/3965cce5c8a5265a4f713a93a7acb2d92b0b72c7/test/memkind_memtier_data_movement_test.cpp#L36)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

Is there any reason behind choosing this particular operation ?

I'm asking mostly out of curiosity
</blockquote></details>

no :)
___
*[test/memkind_memtier_data_movement_test.cpp, line 1548 at r1](https://reviewable.io/reviews/memkind/memkind/785#-MzKzIrq6l8WkFf4JxEF-r1-1548:-MzP7PHM8bf62h3aDGg3:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/3965cce5c8a5265a4f713a93a7acb2d92b0b72c7/test/memkind_memtier_data_movement_test.cpp#L1548)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

+1
</blockquote></details>

Done.
___
*[test/memkind_memtier_data_movement_test.cpp, line 1560 at r1](https://reviewable.io/reviews/memkind/memkind/785#-MzKzIrq6l8WkFf4JxEF-r1-1560:-MzP7QJbBPIYFxpZ2ilJ:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/3965cce5c8a5265a4f713a93a7acb2d92b0b72c7/test/memkind_memtier_data_movement_test.cpp#L1560)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

We don't really use stdout in these unit tests.. Maybe log_info would be a better choice, to avoid stdout clutter?
</blockquote></details>

Done.
___
*[test/memkind_memtier_data_movement_test.cpp, line 1606 at r1](https://reviewable.io/reviews/memkind/memkind/785#-MzKzIrq6l8WkFf4JxEF-r1-1606:-MzP7RLJ2FgkUwKO76Ko:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/3965cce5c8a5265a4f713a93a7acb2d92b0b72c7/test/memkind_memtier_data_movement_test.cpp#L1606)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

Maybe add some stronger assumption ?

Even if no page movement occurred, this test would still pass 50% of the time...

How about:
```C
 ASSERT_TRUE(dram_time2+dram_time1 < pmem_time);
```
 
 
</blockquote></details>

Done.

**Review by bratpiorka — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/785)* status: all files reviewed, 6 unresolved discussions (waiting on @maciekpac and @PatKamin)

**Review by PatKamin — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/785)* status: all files reviewed, 5 unresolved discussions (waiting on @maciekpac)

**Review by maciekpac — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/785)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @bratpiorka)

**Pull request merged**